### PR TITLE
define app 0.6

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

releasing App 0.5.x shall be cut from master fust before this commit

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @tchaton @rohitgr7

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -95,11 +95,14 @@ repos:
...
+                src/pytorch_lightning/CHANGELOG.md|
+                src/lightning_app/CHANGELOG.md
+            )$
 
   - repo: https://github.com/PyCQA/flake8
     rev: 4.0.1
     hooks:
       - id: flake8
         name: Check PEP8
-        exclude: docs/source-app
```
Comment (manskx) [inline .pre-commit-config.yaml:105]: @tchaton This was added on purpose to skip pre-commit on docs ?

Comment (Borda) [reply .pre-commit-config.yaml:105]: well, he wanted to not break his short line code examples, but still, we shall follow PEP8 and this just raises warning does not change anything automatically

**Review by manskx — APPROVED** [review]

File: docs/source-app/examples/github_repo_runner/app.py [hunk 2]
```diff
@@ -198,7 +198,12 @@ def page_1__create_new_run(state):
         "Enter a Github Repo URL", value="https://github.com/Lightning-AI/lightning-quick-start.git"
     )
 
-    default_script_args = "--trainer.max_epochs=5 --trainer.limit_train_batches=4 --trainer.limit_val_batches=4 --trainer.callbacks=ModelCheckpoint --trainer.callbacks.monitor=val_acc"
+    default_script_args = \
+        "--trainer.max_epochs=5" \
+        " --trainer.limit_train_batches=4" \
+        " --trainer.limit_val_batches=4" \
+        " --trainer.callbacks=ModelCheckpoint" \
+        " --trainer.callbacks.monitor=val_acc"
```
Comment (rohitgr7) [inline docs/source-app/examples/github_repo_runner/app.py:206]:
```suggestion
    default_script_args = (
        "--trainer.max_epochs=5"
        " --trainer.limit_train_batches=4"
        " --trainer.limit_val_batches=4"
        " --trainer.callbacks=ModelCheckpoint"
        " --trainer.callbacks.monitor=val_acc"
    )
```

File: README.md [hunk 3]
```diff
@@ -360,7 +360,7 @@ ______________________________________________________________________
 ## Lightning Lite
 
 <div align="center">
-  <img src="docs/source/_static/images/lightning_lite/lite.gif" height="200px" width="600px">
+  <img src="docs/source-pytorch/_static/images/lightning_lite/lite.gif" height="200px" width="600px">
 </div>
 
 In the Lighting 1.5 release, LightningLite now enables you to leverage all the capabilities of PyTorch Lightning Accelerators without any refactoring to your training loop. Check out the
```
Comment (rohitgr7) [inline README.md:366]:
```suggestion
In the Lightning v1.5 release, LightningLite now enables you to leverage all the capabilities of PyTorch Lightning Accelerators without any refactoring to your training loop. Check out the
```

**Review by rohitgr7 — APPROVED** [review]
nit

File: docs/source-app/examples/file_server/app.py [hunk 4]
```diff
@@ -3,9 +3,7 @@
 import tarfile
 import uuid
 import zipfile
-from dataclasses import dataclass
```
Comment (tchaton) [inline docs/source-app/examples/file_server/app.py:6]: Let's not add noqa in this file. This is a doc file and users don't want to see them.

File: docs/source-app/examples/github_repo_runner/app.py [hunk 5]
```diff
@@ -198,7 +198,13 @@ def page_1__create_new_run(state):
         "Enter a Github Repo URL", value="https://github.com/Lightning-AI/lightning-quick-start.git"
     )
 
-    default_script_args = "--trainer.max_epochs=5 --trainer.limit_train_batches=4 --trainer.limit_val_batches=4 --trainer.callbacks=ModelCheckpoint --trainer.callbacks.monitor=val_acc"
+    default_script_args = (
```
Comment (tchaton) [inline docs/source-app/examples/github_repo_runner/app.py:201]: The re-formatting is breaking the docs. Now, we need to re-aligned the cut for the tutorials. Can you re-align them ?

Comment (Borda) [reply docs/source-app/examples/github_repo_runner/app.py:201]: well, on one side you say you format to have short line, but this was too long

File: docs/source-app/examples/file_server/app.py (hunk 4 above)
Comment (Borda) [reply docs/source-app/examples/file_server/app.py:6]: if you do not use noqua, follow python standards ;)

File: src/lightning/__version__.py [hunk 6]
```diff
@@ -1 +1 @@
-version = "20220621"
+version = "20220718"
```
Comment (Borda) [inline src/lightning/__version__.py:1]:
```suggestion
version = "2022.7.18"
```

**Pull request merged**